packages/cw721/src/msg.rs

```diff
@@ -21,15 +60,925 @@ pub enum Cw721ExecuteMsg {
+                return Err(Cw721ContractError::CollectionDescriptionTooLong {
+                    max_length: MAX_COLLECTION_DESCRIPTION_LENGTH,
+                });
+            }
+        }
+
+        // check images are URLs
+        if let Some(image) = &self.image {
+            Url::parse(image)?;
+        }
+        if let Some(external_link) = &self.external_link {
+            Url::parse(external_link)?;
+        }
+        // no need to check royalty info, as it is checked during creation of RoyaltyInfo
+        Ok(())
+    }
+}
+
+#[cw_serde]
+// This is both: a query response, and incoming message during instantiation and execution.
+pub struct RoyaltyInfoResponse {
+    pub payment_address: String,
+    pub share: Decimal,
+}
+
+impl Cw721CustomMsg for RoyaltyInfoResponse {}
+
+impl StateFactory<RoyaltyInfo> for RoyaltyInfoResponse {
+    fn create(
+        &self,
+        deps: Option<Deps>,
+        env: Option<&Env>,
+        info: Option<&MessageInfo>,
+        current: Option<&RoyaltyInfo>,
+    ) -> Result<RoyaltyInfo, Cw721ContractError> {
+        self.validate(deps, env, info, current)?;
+        let deps = deps.ok_or(Cw721ContractError::NoDeps)?;
+        match current {
+            // Some: update existing royalty info
+            Some(current) => {
+                let mut updated = current.clone();
+                updated.payment_address = deps.api.addr_validate(self.payment_address.as_str())?;
+                updated.share = self.share;
+                Ok(updated)
+            }
+            // None: create new royalty info
+            None => {
+                let new = RoyaltyInfo {
+                    payment_address: deps.api.addr_validate(self.payment_address.as_str())?,
+                    share: self.share,
+                };
+                Ok(new)
+            }
+        }
+    }
+
+    fn validate(
+        &self,
+        _deps: Option<Deps>,
+        _env: Option<&Env>,
+        _info: Option<&MessageInfo>,
+        current: Option<&RoyaltyInfo>,
+    ) -> Result<(), Cw721ContractError> {
+        if let Some(current_royalty_info) = current {
+            // check max share delta
+            if current_royalty_info.share < self.share {
+                let share_delta = self.share.abs_diff(current_royalty_info.share);
+
+                if share_delta > Decimal::percent(MAX_ROYALTY_SHARE_DELTA_PCT) {
+                    return Err(Cw721ContractError::InvalidRoyalties(format!(
+                        "Share increase cannot be greater than {MAX_ROYALTY_SHARE_DELTA_PCT}%"
+                    )));
+                }
+            }
+        }
+        // check max share
+        if self.share > Decimal::percent(MAX_ROYALTY_SHARE_PCT) {
+            return Err(Cw721ContractError::InvalidRoyalties(format!(
+                "Share cannot be greater than {MAX_ROYALTY_SHARE_PCT}%"
+            )));
+        }
+        Ok(())
+    }
+}
```

Comment:
@humanalgorithm look here in `RoyaltyInfoResponse` it implements `StateFactory`'s create and validate function. In this case it checks for `MAX_ROYALTY_SHARE_DELTA_PCT` and `MAX_ROYALTY_SHARE_PCT`.